client/src/components/autosave/index.js

```diff
@@ -52,10 +52,13 @@ class Autosave extends React.Component {
 
   statusMessage() {
     const { error, saveInProgress } = this.props;
+
+    if (saveInProgress) return `Saving...`;
```

Comment:
show `Saving...` first, so we show the correct saving state in the following condition:

1. After a hash fails to save after 3 retries (which populates error state) 
1. We make a new change 
1. Save status is now `Saving...` again instead of `Autosave error: Foo`